### PR TITLE
Replace `Logger` with equivalent @apollo/utils logger interface

### PR DESCRIPTION
**Pull request by trevor-scheer**
The `Logger` interface has been ported over to the `apollo-utils` repo since it's a generic interface used by both server and gateway. This PR migrates `Logger` usages to that one and stops exporting it from `apollo-server-types`.

Also updated references to docs over to the `Logger` README.
Fixes https://github.com/apollographql/apollo-server/issues/6056
Fixes #6262

### CODE REVIEW

File: docs/source/api/apollo-server.mdx [hunk 1]
```diff
@@ -367,11 +367,11 @@ Defaults to `true` unless the `NODE_ENV` environment variable is `production` or
 
 ##### `logger`
 
-[`Logger`](https://github.com/apollographql/apollo-server/blob/main/packages/apollo-server-types/src/index.ts#L178-L184)
+[`Logger`](https://github.com/apollographql/apollo-utils/tree/main/packages/logger)
```
Comment (glasser) [inline docs/source/api/apollo-server.mdx:370]: Maybe link to npmjs instead? (In all links in the PR.)

File: packages/apollo-server-types/src/index.ts [hunk 2]
```diff
@@ -184,14 +185,6 @@ export type GraphQLExecutionResult = {
   extensions?: Record<string, any>;
 };
 
-export type Logger = {
```
Comment (glasser) [inline packages/apollo-server-types/src/index.ts:187]: This seems like a backwards-incompatible change that we shouldn't do on `main`. Instead we should export what we imported?

**Review by glasser — CHANGES_REQUESTED** [review]

File: packages/apollo-server-core/src/__tests__/logger.test.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import { ApolloServerBase } from '../..';
```
Comment (glasser) [inline packages/apollo-server-core/src/__tests__/logger.test.ts:1]: You've moved this test to `@apollo/utils.logger`, so I'd think you'd be able to delete it, as well as the top-level dependencies on the various logging packages.

Comment (trevor-scheer) [reply packages/apollo-server-core/src/__tests__/logger.test.ts:1]: Decided to replace with tests that ensure the logger is correctly used by AS / passed through the request pipeline, but yes - removed the other logger deps! e5ebcff

File: packages/apollo-server-types/src/index.ts (hunk 2 above)
Comment (trevor-scheer) [reply packages/apollo-server-types/src/index.ts:187]: Yeah, good catch. [f3dbe44](https://github.com/apollographql/apollo-server/pull/6299/commits/f3dbe44efaa65653e7917c0c6ecd1438109523f5)

File: docs/source/api/apollo-server.mdx (hunk 1 above)
Comment (trevor-scheer) [reply docs/source/api/apollo-server.mdx]: [a69a065](https://github.com/apollographql/apollo-server/pull/6299/commits/a69a065a0db1ff8f770d9fdcddb2a217a0e49b82)

File: packages/apollo-server-core/src/__tests__/logger.test.ts [hunk 4]
```diff
@@ -1,146 +1,58 @@
...
-    logger.setLevel(loglevel.levels.DEBUG);
-
-    await triggerLogMessage(logger);
-
-    expect(sink).toHaveBeenCalledWith({
-      level: loglevel.levels.DEBUG,
-      message: KNOWN_DEBUG_MESSAGE,
-    });
+    expect(debugSpy).toHaveBeenCalledWith(KNOWN_DEBUG_MESSAGE);
+    expect(defaultLogger.levels).toEqual(loglevel.levels);
```
Comment (glasser) [inline packages/apollo-server-core/src/__tests__/logger.test.ts:29]: maybe instanceof instead

**Review by glasser — APPROVED** [review]

File: packages/apollo-server-core/src/__tests__/logger.test.ts (hunk 4 above)
Comment (trevor-scheer) [reply packages/apollo-server-core/src/__tests__/logger.test.ts:31]: Tried, but can't. Left a comment that explains why I'm asserting this at least.

**Pull request merged**